spec/features/takeaway.rb

```diff
@@ -0,0 +1,105 @@
+# class Takeaway
+#
+#   CHECKOUT_ERROR = "Total cost entered does not match the sum of your order!"
+#   ITEM_ERROR = "This item does not exist!"
+#
+#   attr_reader :menu, :order
+#
+#   def initialize(menu = Menu.new, order = Order.new)
+#     @menu = menu
+#     @order = order
+#   end
+#
+#   def read_menu
+#     @menu.read
+#   end
+#
+#   def order(itm, qty=1)
+#     @menu.dishes[itm].nil? ? (raise ITEM_ERROR) : @order.add_to_basket(itm, qty)
+#   end
+#
+#   def basket_summary
+#     @order.basket_sum(@menu)
+#   end
+#
+#   def total_cost
+#     @total = @order.total_bill(@menu)
+#     "Total Cost: £#{@total}"
+#   end
+#
+#   def checkout(final_cost)
+#     correct_amount?(final_cost) ? (send_text) : (raise CHECKOUT_ERROR)
+#   end
+#
+#   private
+#
+#   def correct_amount?(final_cost)
+#     final_cost == @total
+#   end
+#
+#   def send_text
+#     TextMessenger.send_text
+#   end
+#
+# end
+#
+#
+# class Order
+#
+#   def initialize
+#     @basket = Hash.new(0)
+#   end
+#
+#   def add_to_basket(itm, qty)
+#     @basket[itm] += qty
+#     "#{qty}x #{itm}(s) added to your basket."
+#   end
+#
+#   def basket_sum(menu)
+#     @basket.each do |itm, qty|
+#     print "#{itm} x#{qty} = £#{(menu.dishes[itm]*qty).round(2)}, "
+#     end
+#   end
+#
+#   def total_bill(menu)
+#     total = []
+#     @basket.each{ |itm, qty| total << (menu.dishes[itm]*qty).round(2) }
+#     (total.inject(:+)).round(2)
+#   end
+#
+# end
+#
+#
+# class Menu
+#
+#   attr_reader :dishes
+#
+#   def initialize
+#     @dishes = {"Spring Roll" => 0.99, "Fried Prawn" => 2.99}
+#   end
+#
+#   def read
+#     @dishes.each { |item, cost| print "#{item}: £#{cost}, " }
```

Comment:
Style/AsciiComments: Use only ascii symbols in comments. (https://github.com/bbatsov/ruby-style-guide#english-comments)